resources/pre-upgrade.sh

```diff
@@ -10,65 +10,74 @@ DATABASE_USER=$(doguctl config -e sa-postgresql/username)
 DATABASE_USER_PASSWORD=$(doguctl config -e sa-postgresql/password)
 DATABASE_DB=$(doguctl config -e sa-postgresql/database)
 
+echo "Running pre-upgrade script..."
+if [[ ${FROM_VERSION} == *"5"* ]]; then
+  echo "Upgrade from version ${FROM_VERSION} to ${TO_VERSION} is not supported. Upgrade to version 6.7.7-2 before."
+  exit 1
+fi
+
+# Save extensions folder as it henceforth gets its own volume
+if [[ ${FROM_VERSION} == *"6.7.6-1"* ]]; then
+  mkdir /opt/sonar/data/extensions
+  cp -R /opt/sonar/extensions/* /opt/sonar/data/extensions/
+fi
+
 function sql(){
   PGPASSWORD="${DATABASE_USER_PASSWORD}" psql --host "${DATABASE_IP}" --username "${DATABASE_USER}" --dbname "${DATABASE_DB}" -1 -c "${1}"
   return $?
 }
 
-function add_temporary_admin_user() {
+function add_temporary_admin_user_sonar6() {
   # temporarily create admin user and add to admin groups
   TEMPORARY_ADMIN_USER=${1}
-  sql "INSERT INTO users (login, name, crypted_password, salt, active, external_identity, user_local)
-  VALUES ('${TEMPORARY_ADMIN_USER}', 'Temporary Administrator', 'a373a0e667abb2604c1fd571eb4ad47fe8cc0878', '48bc4b0d93179b5103fd3885ea9119498e9d161b', true, '${TEMPORARY_ADMIN_USER}', true);"
+  HASHED_PW=${2}
+  SALT=${3}
+  sql "INSERT INTO users (login, name, crypted_password, salt, active, external_identity, user_local, is_root, onboarded)
+  VALUES ('${TEMPORARY_ADMIN_USER}', 'Temporary Administrator', '${HASHED_PW}', '${SALT}', true, '${TEMPORARY_ADMIN_USER}', true, true, true);"
   ADMIN_ID_PSQL_OUTPUT=$(PGPASSWORD="${DATABASE_USER_PASSWORD}" psql --host "${DATABASE_IP}" --username "${DATABASE_USER}" --dbname "${DATABASE_DB}" -1 -c "SELECT id FROM users WHERE login='${TEMPORARY_ADMIN_USER}';")
   ADMIN_ID=$(echo "${ADMIN_ID_PSQL_OUTPUT}" | awk 'NR==3' | cut -d " " -f 2)
   if [[ -z ${ADMIN_ID} ]]; then
     # id has only one digit
     ADMIN_ID=$(echo "${ADMIN_ID_PSQL_OUTPUT}" | awk 'NR==3' | cut -d " " -f 3)
   fi
-  echo "Got temporary admin id: ${ADMIN_ID}"
   sql "INSERT INTO groups_users (user_id, group_id) VALUES (${ADMIN_ID}, 1);"
   sql "INSERT INTO groups_users (user_id, group_id) VALUES (${ADMIN_ID}, 2);"
 }
 
+function getSHA1PW() {
+    PW="${1}"
+    SALT="${2}"
+    echo -n "--${SALT}--${PW}--" | sha1sum | awk '{print $1}'
+}
+
 function remove_temporary_admin_user() {
   TEMPORARY_ADMIN_USER=${1}
   sql "DELETE FROM users WHERE login='${TEMPORARY_ADMIN_USER}';"
 }
 
-echo "Running pre-upgrade script..."
-if [[ ${FROM_VERSION} == *"5.6.7"* ]] && [[ ${TO_VERSION} == *"6.7."* ]]; then
-  TEMPORARY_ADMIN_USER="admin"
+if [[ ${FROM_VERSION} == *"6"* ]] && [[ ${TO_VERSION} == *"7.9"* ]]; then
```

Comment:
The comparison of the FROM_VERSION should be more specific. As of now, it also applies to sonar v5.6.6-4, for example.